fdbclient/NativeAPI.actor.cpp

```diff
@@ -3257,6 +3257,33 @@ void Transaction::setupWatches() {
 	}
 }
 
+ACTOR Future<TransactionCommitCostEstimation> estimateCommitCosts(Transaction* self, CommitTransactionRef* transaction) {
+	state MutationRef* it = transaction->mutations.begin();
+	state MutationRef* end = transaction->mutations.end();
+	state TransactionCommitCostEstimation trCommitCosts;
+	for (; it != end; ++it) {
+		if (it->type == MutationRef::Type::SetValue) {
+			trCommitCosts.bytesWrite += it->expectedSize();
+			trCommitCosts.numWrite++;
+		} else if (it->isAtomicOp()) {
+			trCommitCosts.bytesAtomicWrite += it->expectedSize();
+			trCommitCosts.numAtomicWrite++;
+		} else if (it->type == MutationRef::Type::ClearRange) {
+			trCommitCosts.numClear ++;
+			if (deterministicRandom()->random01() < CLIENT_KNOBS->EXPENSIVE_COMMIT_COST_ESTIMATION_FRAC) {
```

Comment:
the knob name "_FRAC" seems confusing.
`EXPENSIVE_COMMIT_COST_ESTIMATION` sounds good enough.